noise/README.md

```diff
@@ -291,6 +295,24 @@ internal libp2p data.
 
 The XX handshake MUST be supported by noise-libp2p implementations.
 
+### Noise Extensions
+
+Since the Noise handshake pattern itself doesn't define any extensibility
+mechanism, this specification defines an extension registry, modeled after
+[RFC 6066](https://www.rfc-editor.org/rfc/rfc6066) (for TLS) and
+[RFC 9000](https://datatracker.ietf.org/doc/html/rfc9000#section-19.21)
+(for QUIC).
+
+This spec currently defines 3 extension code points for the `NoiseExtensions`
+protobuf. Note that this document only defines the code point, and leaves
+it up to the protocol using that code point to define semantics associated
+with that code point.
+
+Code points above 1024 MAY be used for experimentation. Code points up to
+this value MUST be registered in this document before deployment.
```

Comment:
As far as I understand we tried to use [multicodec](https://github.com/multiformats/multicodec/) for all magic numbers in the libp2p ecosystem thus far. If I understand this proposal correctly, it runs somewhat against this convention.

I am not opposed to that, I just think it is worth pointing out in the specification. E.g. a short paragraph of why we are not using multicodec for this.

What do you think @marten-seemann?

Reply:
Good point. I'm not super familiar with the design philosophy behind multicoded. @Stebalien and @lidel, could you help us out here?